src/app/static/src/app/utils.ts

```diff
@@ -173,6 +173,26 @@ export const findResource = (datasetWithResources: any, resourceType: string): D
         outOfDate: false} : null
 };
 
+export const datasetFromMetadata = (id: string, datasets: any[], schemas: ADRSchemas) => {
```

Comment:
This moved from SelectDataset component in order to reuse when re-fetching dataset details for backward compatibility